source/extensions/filters/http/transformation/body_header_transformer.cc

```diff
@@ -24,17 +24,16 @@ void BodyHeaderTransformer::transform(
     json_body["body"] = body.toString();
   }
 
-  json &headers = json_body["headers"];
-  header_map.iterate(
-      [&headers](const Http::HeaderEntry &header) -> Http::HeaderMap::Iterate {
-        headers[std::string(header.key().getStringView())] =
-            std::string(header.value().getStringView());
-        return Http::HeaderMap::Iterate::Continue;
-      });
+  auto headers = std::map<std::string, absl::string_view>{};
+  // multi_value_headers will not be populated if add_request_metadata_ is false
+  auto multi_value_headers = std::map<std::string, std::vector<absl::string_view>>{};
+  parse_headers(header_map, headers, multi_value_headers);
+  json_body["headers"] = headers;
```

Comment:
we assign `headers` to json_body` here, but `headers` only holds values as string views (which may or may not outlive `json_body`. i think this could be a bug, and it would be much safer imo to make `headers` and `multiValueHeaders` maps hold values that are strings instead of string views

Reply:
Done here: https://github.com/solo-io/envoy-gloo/pull/207/commits/922b07899c3bdbe007ef8e0692a1a02f42b6255d ,
 with some additional changes here: https://github.com/solo-io/envoy-gloo/pull/207/commits/3081fa2a55ea0e22ead1787389c6f7452ff8770d